src/matlab/FiniteElement.m

```diff
@@ -282,6 +283,17 @@
                         end
                     end
                 end
+
+                % add user variables
+                z_user = define_casadi_symbolic(settings.casadi_symbolic_mode,...
+                                                ['z_user_' num2str(ctrl_idx-1) '_' num2str(fe_idx-1) '_' num2str(ii)],...
+                                                dims.n_z_user);
+                obj.addVariable(z_user,...
+                                'z_user',...
+                                -inf*ones(dims.n_z_user,1),...
+                                inf*ones(dims.n_z_user,1),...
```

Comment:
I would make it possible for the user to add lbz and ubz box constraints on the algebraic variables